plugins/woocommerce-admin/client/analytics/components/report-chart/index.js

```diff
@@ -91,38 +97,100 @@ export class ReportChart extends Component {
 			query,
 			defaultDateRange
 		);
-		const chartData = primaryData.data.intervals.map( function (
-			interval,
-			index
-		) {
-			const secondaryDate = getPreviousDate(
+
+		const primarydataContainsLeapYear = dataContainsLeapYear( primaryData );
+		const secondarydataContainsLeapYear =
+			dataContainsLeapYear( secondaryData );
+		const primaryDataIntervals = [ ...primaryData.data.intervals ];
+		const secondaryDataIntervals = [ ...secondaryData.data.intervals ];
+
+		const chartData = [];
+
+		for ( let index = 0; index < primaryDataIntervals.length; index++ ) {
+			const interval = primaryDataIntervals[ index ];
+
+			const secondaryDateMoment = getPreviousDate(
```

Comment:
Nit: I think if we move `secondaryDateMoment` to closer to where it's used (line 128), it would be easier to understand the code.

Reply:
Sounds good, it's moved 👍